client/src/main/java/io/pravega/client/state/impl/StateSynchronizerImpl.java

```diff
@@ -175,9 +175,10 @@ public void updateStateUnconditionally(List<? extends Update<StateT>> update) {
 
     @Override
     public void initialize(InitialUpdate<StateT> initial) {
-        Revision result = client.writeConditionally(client.fetchOldestRevision(), new UpdateOrInit<>(initial));
+        Revision result = client.writeConditionally(new RevisionImpl(segment, 0, 0), new UpdateOrInit<>(initial));
```

Comment:
Will it impact anything with compacted state synchronizer stream?

Reply:
If the state is compacted and the oldest revision is non-zero and you call init on a new state synchronizer it will not attempt to overwrite the state with the supplied initial state.